azure-cli-extensions.rdbms-connect: init at 1.0.6

obreitwi opened the pull request:
## Description of changes

* Added the [azure-cli-extensions.rdbms-connect](https://github.com/Azure/azure-cli-extensions/tree/main/src/rdbms-connect) at `v1.0.6`.
* As drive-by, applied `nixfmt-rfc-style` to 
  * `pkgs/tools/admin/azure-cli/default.nix` (in a [separate commit](https://github.com/NixOS/nixpkgs/commit/ae557fe541d6a7c230169c92480d01ad9c4764f3))
  * `pkgs/tools/admin/azure-cli/extensions-manual.nix` 
* Since `rdbms-connect` has its [own dependencies](https://github.com/Azure/azure-cli-extensions/blob/a8a793eb4a88a7045233733573e4ee338fdc0f6d/src/rdbms-connect/setup.py#L38C1-L43C2), this required [forwarding](https://github.com/NixOS/nixpkgs/commit/c06c1b780ae96d3c15e1f840e5cdb0ca3e0e0b78) of `propagatedBuildInputs` of extensions to the base derivation.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

katexochen commented:
Why the reformatting?

obreitwi commented:
> Why the reformatting?

[Was told in a different PR](https://github.com/NixOS/nixpkgs/pull/316730#pullrequestreview-2092616402) that `nixpkgs` is in the process of adapting a new linting style.

Hence I did a reformat beforehand for any file I touched. But, if there are efforts to reformat at once, I can also drop the commit.

Still new to `nixpkgs` and getting used to its customs.

katexochen commented:
> > Why the reformatting?
> 
> [Was told in a different PR](https://github.com/NixOS/nixpkgs/pull/316730#pullrequestreview-2092616402) that `nixpkgs` is in the process of adapting a new linting style.
> 
> Hence I did a reformat beforehand for any file I touched. But, if there are efforts to reformat at once, I can also drop the commit.
> 
> Still new to `nixpkgs` and getting used to its customs.

Such efforts with huge impact are always a bit of pain in nixpkgs. I asked the formatting team before but couldn't get a clear answer to whether individual packages should be formatted or not. I think many maintainers heard of the upcoming change an are just impatient. Thing is reformatting leads to merge conflicts with other PRs.

katexochen commented:
Result of `nixpkgs-review pr 317841` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>azure-cli-extensions.rdbms-connect</li>
    <li>azure-cli-extensions.rdbms-connect.dist</li>
  </ul>
</details>

katexochen commented on pkgs/tools/admin/azure-cli/extensions-manual.nix:
```diff
@@ -9,8 +11,21 @@
+    propagatedBuildInputs = with python3Packages; [ distro ];
+  };
+
+  rdbms-connect = mkAzExtension rec {
+    pname = "rdbms-connect";
```
As these extensions can't be manually updated, could you add yourself as maintainer?

obreitwi replied:
Sure, added myself to the list.

katexochen commented on pkgs/tools/admin/azure-cli/extensions-manual.nix:
```diff
@@ -1,5 +1,7 @@
-{ mkAzExtension
-, python3Packages
+{
+  mkAzExtension,
+  mycli,
+  python3Packages,
 }:
```
This is still formatted RFC style.

obreitwi replied:
Reformatted via `nixfmt-classic`. Btw is that the "official" formatter for `azure-cli` at the moment?

Because right now I get diffs for `nixfmt-rfc-style` (obviously) and `alejandra`, but also nixfmt-classic`.

katexochen replied:
Previously to nixfmt-rfc-style, [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt) was/is the common formatter for nixpkgs. Sorry for the trouble.

obreitwi commented:
> [...] Thing is reformatting leads to merge conflicts with other PRs.

Yeah, dropped the reofmatting-commit when I rebased onto recently merged changes. :ok_hand:

katexochen commented on pkgs/tools/admin/azure-cli/default.nix:
```diff
@@ -233,7 +233,7 @@ py.pkgs.toPythonApplication (py.pkgs.buildAzureCliPackage rec {
     # we're using the default immutable configuration.
     pip
-  ];
+  ] ++ lib.concatMap (extension: extension.propagatedBuildInputs) withExtensions;
```
I'm always confused this isn't happening automatically. Thanks for the fix!

obreitwi replied:
Happy to help!

katexochen reviewed (commented):
I've build the cli with the extension, built successfully and show the expected command in the help.

obreitwi committed:
azure-cli-extensions.rdbms-connect: init at 1.0.6

Signed-off-by: Oliver Breitwieser <oliver@breitwieser.eu>

obreitwi committed:
azure-cli: propagate build inputs of extensions

Signed-off-by: Oliver Breitwieser <oliver@breitwieser.eu>

obreitwi committed:
maintainers: add obreitwi

katexochen commented:
Regarding the commit messages: Could you put the `maintainers: add obreitwi` commit first? So you don't need a separate commit and can squash `chore: add obreitwi as maintainer` into `azure-extensions.rdbms-connect: init at 1.0.6`.
`fix: propagate build inputs of extensions` should be be `azure-cli: propagate build inputs of extensions`.

obreitwi commented:
Adjusted :+1:

katexochen committed:
azure-cli-extensions: allow specifying maintainers

Signed-off-by: Paul Meyer <49727155+katexochen@users.noreply.github.com>

katexochen commented:
@obreitwi Sorry, had to make some more adjustments.

- I've added a commit allowing to set `meta.maintainers` on the extensions. When asking you to add yourself as maintainer, I meant to the extension, not the whole azure-cli package, assuming that was possible, but it wasn't.
- I've moved your entry as maintainer over to the extension (now that this is possible).
- Fixed the formatting, using nixpkgs-fmt for now.
- I've scoped the `with python3Packages;` to a list of only the Python packages, and moved the other external dependency into a separate list. This is just for precaution and style, variables escaping a scope are discouraged (someone could add a mycli package to python packages and the meaning would change).

Thanks for the contribution and sorry for the back and forth. This is the first time someone adds an extension.

obreitwi commented:
> Thanks for the contribution and sorry for the back and forth. This is the first time someone adds an extension.

No problem, thank _you_ for the feedback; always happy to learn :+1:
